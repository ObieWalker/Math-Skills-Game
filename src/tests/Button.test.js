import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Button from '../../src/components/Button';

let mountedForm, props, wrapper;
let NumbersList = {
  list : []
}
const checkAnswer = jest.fn();
const acceptAnswer = jest.fn();
const redraw = jest.fn();
const redraws = true;
let answerIsCorrect = true;
const selectedNumbers = []


const getComponent = () => {
  if (!mountedForm) {
    props = {
      answerIsCorrect,
      acceptAnswer,
      checkAnswer,
      selectedNumbers,
      redraw,
      redraws
    };
    mountedForm = shallow(<Button {...props} />);
  }
  return mountedForm;
};

describe('Button Component', () => {
  beforeEach(() => {});


  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('br').length).toBe(2);
    expect(wrapper.find('button').length).toBe(2);
  });
  it('renders component successfully', () => {
    wrapper = getComponent();
    wrapper.setProps({ answerIsCorrect: false });
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('br').length).toBe(2);
    expect(wrapper.find('button').length).toBe(2);
  });
});
