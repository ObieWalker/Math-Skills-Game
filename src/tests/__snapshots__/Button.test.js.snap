// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    acceptAnswer={[MockFunction]}
    answerIsCorrect={true}
    checkAnswer={[MockFunction]}
    redraw={[MockFunction]}
    redraws={true}
    selectedNumbers={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn btn-success"
          onClick={[MockFunction]}
        >
          <i
            className="fa fa-check"
          />
        </button>,
        <br />,
        <br />,
        <button
          className="btn btn-warning btn-sm"
          disabled={false}
          onClick={[MockFunction]}
        >
          <i
            className="fa fa-refresh"
          >
            reload 
            true
          </i>
        </button>,
      ],
      "className": "col-2",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-check"
          />,
          "className": "btn btn-success",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-check",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa fa-refresh"
          >
            reload 
            true
          </i>,
          "className": "btn btn-warning btn-sm",
          "disabled": false,
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "reload ",
              true,
            ],
            "className": "fa fa-refresh",
          },
          "ref": null,
          "rendered": Array [
            "reload ",
            true,
          ],
          "type": "i",
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn btn-success"
            onClick={[MockFunction]}
          >
            <i
              className="fa fa-check"
            />
          </button>,
          <br />,
          <br />,
          <button
            className="btn btn-warning btn-sm"
            disabled={false}
            onClick={[MockFunction]}
          >
            <i
              className="fa fa-refresh"
            >
              reload 
              true
            </i>
          </button>,
        ],
        "className": "col-2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-check"
            />,
            "className": "btn btn-success",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-check",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-refresh"
            >
              reload 
              true
            </i>,
            "className": "btn btn-warning btn-sm",
            "disabled": false,
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "reload ",
                true,
              ],
              "className": "fa fa-refresh",
            },
            "ref": null,
            "rendered": Array [
              "reload ",
              true,
            ],
            "type": "i",
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
